libs/azure-storage/tests/unit_tests/test_document_loaders.py

```diff
@@ -0,0 +1,160 @@
+
+
+@pytest.fixture
+def blob_names() -> list[str]:
+    return ["text_file.txt", "json_file.json", "csv_file.csv"]
+
+
+@pytest.fixture
+def blob_contents() -> list[str]:
+    return [
+        "test content",
+        "{'test': 'test content'}",
+        "col1,col2\nval1,val2",
+    ]
+
+
+def _blob_storage_loader_args() -> dict:
+    return {
+        "account_url": "https://testaccount.blob.core.windows.net",
+        "container_name": "test-container",
+    }
+
+
+@pytest.fixture
+def mock_lazy_load_documents_from_blob(
+    mocker: MockerFixture, blob_names: list[str], blob_contents: list[str]
+) -> None:
+    from langchain_core.documents.base import Document
+
+    docs = []
+    for name, content in zip(blob_names, blob_contents):
+        docs.append(
+            Document(
+                page_content=content,
+                metadata={
+                    "source": f"https://testaccount.blob.core.windows.net/test-container/{name}"
+                },
+            )
+        )
+    mocker.patch(
+        "langchain_azure_storage.document_loaders.AzureBlobStorageLoader._lazy_load_documents_from_blob",
+        return_value=iter(docs),
+    )
+
+
+def test_lazy_load(
+    mocker: MockerFixture,
+    blob_names: list[str],
+    blob_contents: list[str],
+    mock_lazy_load_documents_from_blob: None,
+) -> None:
+    mock_container_client = mocker.patch.object(ContainerClient, "from_container_url")
+    mock_client = mocker.MagicMock()
+    mock_client.list_blob_names.return_value = blob_names
+    mock_container_client.return_value = mock_client
+
+    args = {**_blob_storage_loader_args(), "blob_names": blob_names}
+    loader = AzureBlobStorageLoader(**args)
+
+    for doc, expected_content, blob_name in zip(
+        loader.lazy_load(), blob_contents, blob_names
+    ):
+        assert doc.page_content == expected_content
+        assert (
+            doc.metadata["source"]
+            == f"https://testaccount.blob.core.windows.net/test-container/{blob_name}"
+        )
+
+
+def test_get_blob_client(
+    mocker: MockerFixture,
+    mock_lazy_load_documents_from_blob: None,
+) -> None:
+    mock_container_client = mocker.patch.object(ContainerClient, "from_container_url")
+    mock_client = mocker.MagicMock()
+    mock_client.list_blob_names.return_value = ["text_file.txt"]
+    mock_container_client.return_value = mock_client
+    mock_blob_client = mock_client.get_blob_client
+
+    args = {**_blob_storage_loader_args(), "prefix": "text"}
+    loader = AzureBlobStorageLoader(**args)
+    list(loader.lazy_load())
+    mock_blob_client.assert_called_once_with("text_file.txt")
+
+
+def test_default_credential(mocker: MockerFixture) -> None:
+    mock_default_credential = mocker.patch.object(
+        azure.identity, "DefaultAzureCredential"
+    )
+    mock_default_credential.return_value = mocker.MagicMock()
+    mock_container_client = mocker.patch.object(ContainerClient, "from_container_url")
```

Comment:
Instead of patching as part of the test, it might make sense to just apply an auto-fixture to patch the ContainerClient class such as we do [here in the PyTorch connector](https://github.com/Azure/azure-storage-for-pytorch/blob/40cd9b66eaa0a9214553dc7f4e69ebf813063e5e/tests/unit/test_client.py#L301-L315). Mainly that will guarantee that we never try to make API calls as part of the unit tests and also help reduce some of the verbosity of the test case bodies.